flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1079,6 +1079,9 @@ class JobManager(
           executionGraph.registerExecutionListener(gateway)
           executionGraph.registerJobStatusListener(gateway)
         }
+
+        // All good. Submission succeeded!
+        jobInfo.client ! decorateMessage(JobSubmitSuccess(jobGraph.getJobID))
```

Comment:
Hmm but now we have the problem that the user might see a `JobSubmitSuccess` without the job being stored in the `SubmittedJobGraphStore`, right? This means that if the `JobManager` dies before the job is persisted, it will never be recovered. I think this violates the `JobSubmitSuccess` contract.